chore(deps): upgrade tinybench to v6

jerome-benoit opened the pull request:
### Description

Upgrade tinybench from v2.9.0 to v6.0.0.

**Breaking changes adapted:**
- `TaskResult` is now a discriminated union — `BenchmarkResult` redefined as standalone vitest-owned interface with `BenchmarkStatistics` type and `createEmptyStatistics()` helper for type-safe defaults
- Deprecated top-level fields removed (`hz`, `samples`, `mean`, etc.) — using `latency.*`, `throughput.*` instead
- `retainSamples` option (defaults to `false`) replaces manual `samples.length = 0` clearing — mapped to vitest's `includeSamples` via centralized `getBenchOptions`
- `BenchEvent` is now a typed class — event handlers use typed event API
- `BenchmarkResult.sampleCount` renamed to `samplesCount` to align with `Statistics.samplesCount`

**Bug fixes included:**
- Fix `complete` event handler crash when task errors — tinybench v6 fires `complete` after `error`, but errored tasks have no `latency` property. Added task state guard and error tracking
- Fix errored benchmarks silently marked as `pass` — now correctly marked as `fail`
- Replace `Object.assign(result, task.result)` with selective field extraction — prevents tinybench runtime metadata (`runtime`, `runtimeVersion`, `timestampProviderName`) from polluting `BenchmarkResult` and JSON output
- Fix comparison display `diffFixed` check (`'1.0.0'` → `'1.00'`) with proper `else if` chain

**User-facing type change:** `BenchmarkResult.sampleCount` → `BenchmarkResult.samplesCount`. `BenchTaskResult` re-export is now the tinybench v6 `TaskResult` discriminated union.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.
- [x] Please check [Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to make review process faster.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner.

jerome-benoit committed:
chore(deps): update tinybench to 3.x.x

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>

jerome-benoit committed:
fix: import the proprer FnOptions tinybench type

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>

jerome-benoit committed:
fix: import the right BenchOptions type

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>

jerome-benoit committed:
refactor: remove console.log debugging

Signed-off-by: Jérôme Benoit <jerome.benoit@piment-noir.org>

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit commented:
It's still on my TODO list, but I had no time to finish the migration.

gperdomor commented:
@jerome-benoit thank you for the quick response... I hope you find some time to finish 👍🏻

jerome-benoit committed:
Merge remote-tracking branch 'upstream/main' into chore/tinybench-upgrade

# Conflicts:
#	packages/vitest/src/runtime/runners/benchmark.ts
#	test/benchmark/test/reporter.test.ts

jerome-benoit committed:
fix: adapt benchmark reporter and runner for tinybench 3 API after upstream merge

- Use latency.mean instead of deprecated mean in benchmark ranking sort
- Guard filter with benchmark?.latency to handle in-progress benchmarks
- Simplify error handler with null-safe access (e.task?.result?.error)
- Remove dead test:benchmark script referencing deleted workspace

jerome-benoit committed:
chore(deps): upgrade tinybench to v6

jerome-benoit commented on packages/vitest/package.json:
```diff
@@ -188,7 +188,7 @@
     "std-env": "catalog:",
-    "tinybench": "^2.9.0",
+    "tinybench": "^6.0.0",
     "tinyexec": "^1.0.2",
```
Fixed — PR description now correctly states v2.9.0 → v6.0.0.

jerome-benoit committed:
fix: address review comments — type-safe defaults, remove unused array, fix comparison display

jerome-benoit committed:
fix: handle errored benchmark tasks and prevent result pollution

Guard complete handler against errored/aborted tasks to prevent TypeError crash. Track error state to correctly mark failed benchmarks. Replace Object.assign with selective field extraction to prevent tinybench runtime metadata from leaking into BenchmarkResult and JSON output.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)

Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
refactor: centralize bench options mapping in getBenchOptions

Move retainSamples (includeSamples → retainSamples) mapping from the runner call site into getBenchOptions, which now takes the runner config and returns ready-to-use tinybench BenchOptions.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)

Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
refactor: align BenchmarkResult.samplesCount with tinybench naming

Rename numberOfSamples to samplesCount to match tinybench Statistics.samplesCount.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-opencode)

Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit commented:
> By accepting this PR we might introduce bugs that need to be fixed in an already kind of softly deprecated feature, so I would rather avoid any turbulence at this time.

Fair enough. But given the actual minimal code churn and the obviousness of the trivial fixes (data structure fields rename, proper condition and code flow, grouped steps, ...  and all of it relying on a library version that does not has single bug report since), I do not see how it be worse with it.  The current implementation is buggy. Making it just works better will not introduce any maintenance overhead, at all, unless the revamp has already started and that PR will introduce conflicts in another PR.

jerome-benoit commented:
> All of these just increases maintenance, someone needs to write write this, then review this, triage the issue to see if it needs to be reverted. 

Discouraging external contributions made by people having a clue on what they are doing is a very bad move for a project maintenance point of view. Especially on a low priority and under maintained component orthogonal to the project main use case. And is even worse if the proposal is offering a baseline unit tests integration that will establish a solid TDD for any future enhancements. 
If your goal was to make potential valuable contribution to the project go away, it's a complete success :)

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

jerome-benoit committed:
Merge branch 'main' into chore/tinybench-upgrade

sheremet-va commented:
Hey! Sorry for keeping you waiting. I incorporated some of your changes in https://github.com/vitest-dev/vitest/pull/10113 ([docs](https://deploy-preview-10113--vitest-dev.netlify.app/guide/benchmarking)). Would love to know your opinion on the proposed design.

jerome-benoit commented:
> Hey! Sorry for keeping you waiting. I incorporated some of your changes in #10113 ([docs](https://deploy-preview-10113--vitest-dev.netlify.app/guide/benchmarking)). Would love to know your opinion on the proposed design.

An PR sync automation was updating that PR. I guess I can close it. I will look at the other when time permits.
